lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -671,6 +673,37 @@ def add_network_tokenization_card(xml, payment_method, options)
         end
       end
 
+      def add_encrypted_wallet(xml, payment_method)
+        source = payment_method.source == :apple_pay ? 'APPLEPAY' : 'PAYWITHGOOGLE'
```

Comment:
So a customer can't send a source, that is determined by the code base that is using ActiveMerchant. 

That being said I thing you are still right that I should check explicitly for Google Pay as well and raise an error if it is neither, in case there is another encrypted wallet added in the future that we need to support.